modules/fedimint-lnv2-tests/src/bin/self_payment.rs

```diff
@@ -90,6 +92,64 @@ async fn main() -> anyhow::Result<()> {
             serde_json::to_value(FinalReceiveState::Claimed).expect("JSON serialization failed"),
         );
 
+        // test self payment with two gateways
+
+        let gw_cln = dev_fed.gw_cln().await?;
```

Comment:
Going forward we should consider writing these tests agnostic of the lightning node type. We'll need this for LDK as well